homeassistant/components/elmax/alarm_control_panel.py

```diff
@@ -74,31 +83,64 @@ async def async_alarm_arm_away(self, code: str | None = None) -> None:
                 f"Cannot arm {self.name}: please check for open windows/doors first"
             )
 
-        await self.coordinator.http_client.execute_command(
-            endpoint_id=self._device.endpoint_id,
-            command=AreaCommand.ARM_TOTALLY,
-            extra_payload={"code": code},
-        )
-        await self.coordinator.async_refresh()
+        self._pending_state = STATE_ALARM_ARMING
+        self.async_write_ha_state()
+
+        try:
+            await self.coordinator.http_client.execute_command(
+                endpoint_id=self._device.endpoint_id,
+                command=AreaCommand.ARM_TOTALLY,
+                extra_payload={"code": code},
+            )
+        except ElmaxApiError as err:
+            raise HomeAssistantError(
+                "Failed to arm the alarm. An API error occurred."
```

Comment:
Ideally, we should add translations for this, see:

https://developers.home-assistant.io/docs/internationalization/core/#exceptions

Reply:
Okay! Handled in 8ef8aa95b1.